packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -1047,6 +1050,66 @@ export function authorizationServiceBuilder(
 
       return updatedProducerKeychain;
     },
+    async removeProducerKeychainKeyById({
+      producerKeychainId,
+      keyIdToRemove,
+      authData,
+      correlationId,
+      logger,
+    }: {
+      producerKeychainId: ProducerKeychainId;
+      keyIdToRemove: string;
+      authData: AuthData;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<void> {
+      logger.info(
+        `Removing key ${keyIdToRemove} from producer keychain ${producerKeychainId}`
+      );
+
+      const producerKeychain = await retrieveProducerKeychain(
+        producerKeychainId,
+        readModelService
+      );
+      assertOrganizationIsProducerKeychainProducer(
+        authData.organizationId,
+        producerKeychain.data
+      );
+
+      if (
+        authData.userRoles.includes(userRoles.SECURITY_ROLE) &&
+        !producerKeychain.data.users.includes(authData.userId)
+      ) {
+        throw userNotAllowedOnProducerKeychain(
```

Comment:
I may be wrong, unless the user is an admin, the key can be deleted only by the operator that owns it

Notes: 
- this behaviour is not implemented in Scala, so I'll approve anyway this PR
- the same considerations are valid for `deleteClientKeyById`